sklearn/utils/extmath.py

```diff
@@ -1208,14 +1208,10 @@ def stable_cumsum(arr, axis=None, rtol=1e-05, atol=1e-08):
     out : ndarray
         Array with the cumulative sums along the chosen axis.
     """
-    xp, _ = get_namespace(arr)
```

Comment:
It looks like `stable_cumsum` is used in other places as well. This makes me wonder if "always convert to numpy" is a good choice for all callers. To answer this we'd have to be sure that converting to Numpy makes no/little difference to the performance.

An alternative would be to use something like the `max_precision_float_dtype` from #27232 to handle the "no flaot64 on MPS". I think the need for "highest precision float' comes up in a few PRs, so maybe it is a good idea to have it?

WDYT?

Reply:
It is a pragmatic solution for `PCA` tough. So maybe we go with it for now and "re-Array API-ify" it when we deal with another caller where it is needed?

Reply:
Note that we don't have the regular `xp.searchstored` in the standard either, and even if some library implement it it's not always working consistently (e.g. in CuPy it only works for integers arguments, not floats). So even if we had a way do deal with the floating point precision level problem as you suggest, this wouldn't be enough for PCA. So I think for PCA where we need both a xp.cumsum and a xp.searchsorted that work on float values, converting to numpy is the least disruptive/most maintainable solution. Furthermore it's not a large data-structure (it does not grow with `n_samples`, just `n_features`) and it's not in a performance critical section of the code (the SVD part).